functions/src/menu/menu.js

```diff
@@ -0,0 +1,117 @@
+exports.createMenu = function createMenu(values = []) {
```

Comment:
Is it possible to use a TypeScript for the functions?

Reply:
It's possible, but it complicates the matter. I chose plain JS intentionally to avoid any unnecessary build/pre-build steps. Typescript is a useful tool but, IMHO, it starts to shine only in large scale apps. Small bits of code like above should do fine in plain JS. What do you think?